### PR TITLE
Add a section about using the state layer and move controller specifi…

### PR DESCRIPTION
**Pull request by laevandus**
…c content to its own page. Add using controllers link to the sidebar

### 🔗 Issue Links

Related [ios-issues-tracking/728](https://github.com/GetStream/ios-issues-tracking/issues/728)

### 🎯 Goal

Addressing @nuno-vieira suggestions

Screenshot 2024-05-06 at 15 07 22
Screenshot 2024-05-06 at 15 07 28
Screenshot 2024-05-06 at 15 07 32

### ☑️ Contributor Checklist

- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] This change follows zero ⚠️ policy (required)
- [ ] This change should be manually QAed
- [ ] Changelog is updated with client-facing changes
- [ ] New code is covered by unit tests
- [ ] Comparison screenshots added for visual changes
- [ ] Affected documentation updated (docusaurus, tutorial, CMS)

### CODE REVIEW

File: docusaurus/docs/iOS/client/controllers/controllers-overview.md [hunk 1]
```diff
@@ -26,3 +23,38 @@ It's a best practice to pass synchronize a completion block to handle error case
     }
     /// ...
 ```
+
+## About `synchronize()`
```
Comment (laevandus) [inline docusaurus/docs/iOS/client/controllers/controllers-overview.md:27]: Cut & paste from llc overview

File: docusaurus/docs/iOS/client/llc-client-overview.md [hunk 2]
```diff
@@ -9,61 +9,14 @@ Both our [UIKit](../uikit/getting-started.md) and [SwiftUI](../swiftui/getting-s
...
 
 Each controller exposes API functionality and supports delegation. Controllers and their delegates are documented here based on the kind of data they control and allow you to observe. You can find examples on how to build your own view as well.
 
-In addition to controllers, we provide a modern [state layer](state-layer/state-layer-overview.md) with async-await and observable state objects.
+## Accessing State with State Layer
 
-## Using Controllers
+In addition to controllers, we provide a modern [state layer](state-layer/state-layer-overview.md). It follows an architecture where we have objects interacting with the Stream API through async functions. These objects are accompanied with state objects that hold the current state.
 
-Controllers expose functionality via event subscriptions and API calls. In most cases you need to call the `synchronize` method on your controller instance. This is done to avoid side-effects from initialization and to make sure that the controller is ready to be used.
```
Comment (laevandus) [inline docusaurus/docs/iOS/client/llc-client-overview.md:20]: Moved to another page just about controllers

**Review by martinmitrevski — APPROVED** [review]

File: docusaurus/docs/iOS/client/llc-client-overview.md [hunk 3]
```diff
@@ -9,61 +9,14 @@ Both our [UIKit](../uikit/getting-started.md) and [SwiftUI](../swiftui/getting-s
...
-The `StreamChat` framework comes with controllers and delegates that you can use to build your own views.
+The `StreamChat` framework comes with [controllers and delegates](controllers/controllers-overview.md) that you can use to build your own views.
 
 Each controller exposes API functionality and supports delegation. Controllers and their delegates are documented here based on the kind of data they control and allow you to observe. You can find examples on how to build your own view as well.
 
-In addition to controllers, we provide a modern [state layer](state-layer/state-layer-overview.md) with async-await and observable state objects.
+## Accessing State with State Layer
 
-## Using Controllers
+In addition to controllers, we provide a modern [state layer](state-layer/state-layer-overview.md). It follows an architecture where we have objects interacting with the Stream API through async functions. These objects are accompanied with state objects that hold the current state.
```
Comment (nuno-vieira) [inline docusaurus/docs/iOS/client/llc-client-overview.md:18]: The only thing weird here is that we talk about the controllers, then the state layer, and then back to the controllers 🤔 

Ideally, we would talk first about the State Layer and saying it is the new way of managing state, and only then we talk about the controllers

Comment (laevandus) [reply docusaurus/docs/iOS/client/llc-client-overview.md]: Changed it a bit.

**Pull request merged**